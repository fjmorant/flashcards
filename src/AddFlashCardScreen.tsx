import gql from 'graphql-tag'
import moment from 'moment'
import * as React from 'react'
import {graphql} from 'react-apollo'
import {
  Alert,
  Button as ReactNativeButton,
  StyleSheet,
  View,
} from 'react-native'
import * as shortid from 'shortid'
import * as request from 'superagent'
import Button from './common/Button'
import Input from './common/Input'
import InputArea from './common/InputArea'
import ModalPicker from './common/ModalPicker'

import {flashcardsQuery} from './queries'

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 8,
  },
  input: {
    margin: 4,
    flex: 1,
    fontSize: 19,
  },
  inputContainer: {flexDirection: 'row', alignItems: 'center'},
  buttonContainer: {margin: 5},
})

export interface IProps {
  navigation: any
  flashCardList: any
  mastered: boolean
  mutate: any
}

export interface IState {
  modalVisible: boolean
  pickerOptions: Array<{
    definition: string
    examples: Array<string>
  }>
  name: string
  meaning: string
  example: string
  mastered: boolean
}

export class AddFlashCardScreen extends React.Component<IProps, IState> {
  public static navigationOptions = () => ({
    title: 'Add Flash Cards',
    headerRight: null,
  })

  constructor(props: IProps) {
    super(props)

    this.state = {
      modalVisible: false,
      pickerOptions: [],
      name: '',
      meaning: '',
      example: '',
      mastered: false,
    }
  }

  public shouldEnableSaveButton() {
    return this.state.name && this.state.meaning && this.state.example
  }

  public changeName = (name: string) => {
    this.setState({
      name,
    })
  }

  public changeMeaning = (meaning: string) => {
    this.setState({
      meaning,
    })
  }

  public changeExample = (example: string) => {
    this.setState({
      example,
    })
  }

  public onSearchEntryTriggered = () => {
    if (this.state.name) {
      const name = this.state.name.toLowerCase()

      request
        .get(`https://wordsapiv1.p.mashape.com/words/${name}`)
        .set(
          'X-Mashape-Key',
          'VQa8Eayg2Kmsh0nklI2QUhfltYkvp19WrI8jsnxrvGVatb8Lf8'
        )
        .set('Accept', 'application/json')
        .end((error, response) => {
          if (error) {
            Alert.alert(error.message)
            return
          }

          if (response.body.results.length === 1) {
            const entryDictionary = response.body.results[0]
            this.changeMeaning(entryDictionary.definition)
            if (entryDictionary.examples && entryDictionary.examples[0]) {
              this.changeExample(entryDictionary.examples[0])
            }
          } else {
            this.setState({
              pickerOptions: response.body.results,
              modalVisible: true,
            })
          }
        })
    }
  }

  public onSelectMeaningOption = (entry: {key: number}) => {
    const entryDictionary = this.state.pickerOptions[entry.key]
    const hasExample = entryDictionary.examples && entryDictionary.examples[0]

    this.changeMeaning(entryDictionary.definition)
    this.changeExample(hasExample ? entryDictionary.examples[0] : '')

    this.setState({
      pickerOptions: [],
    })
  }

  public clearPickerOptions = () => {
    this.setState({
      pickerOptions: [],
      modalVisible: false,
    })
  }

  public onPressSaveButton = async () => {
    try {
      await this.props.mutate({
        variables: {
          name: this.state.name,
          meaning: this.state.meaning,
          example: this.state.example,
          mastered: false,
        },
        update: (proxy: any) => {
          const data = proxy.readQuery({query: flashcardsQuery})
          const newFlashcards = [
            {
              __typename: 'Flashcard',
              createdAt: moment().toISOString(),
              id: shortid.generate(),
              name: this.state.name,
              meaning: this.state.meaning,
              example: this.state.example,
              mastered: false,
            },
            ...data.User.flashcards,
          ]

          proxy.writeQuery({
            query: flashcardsQuery,
            data: {
              ...data,
              User: {
                ...data.User,
                flashcards: newFlashcards,
              },
            },
          })
        },
      })

      this.props.navigation.goBack()
    } catch (error) {
      // tslint:disable-next-line:no-console
      console.log(error)
      alert('Error saving flashcard')
    }
  }

  public render() {
    return (
      <View style={styles.container}>
        {this.state.pickerOptions.length > 0 ? (
          <ModalPicker
            data={this.state.pickerOptions.map((result, index) => ({
              key: index,
              label: result.definition,
            }))}
            initValue="Select part speech"
            modalVisible={this.state.modalVisible}
            onChange={this.onSelectMeaningOption}
            onClose={this.clearPickerOptions}
          />
        ) : null}
        <View style={styles.inputContainer}>
          <Input
            height={35}
            onChangeText={this.changeName}
            placeholder="Type word you want to remember"
            style={styles.input}
            value={this.state.name}
          />
          <ReactNativeButton
            onPress={this.onSearchEntryTriggered}
            title={'Find'}
          />
        </View>
        <InputArea
          onChangeText={this.changeMeaning}
          placeholder="Explanation of the word"
          style={styles.input}
          value={this.state.meaning}
        />
        <InputArea
          onChangeText={this.changeExample}
          placeholder="Example of your word"
          style={styles.input}
          value={this.state.example}
        />
        <View style={styles.buttonContainer}>
          <Button
            disabled={!this.shouldEnableSaveButton()}
            height={40}
            onPress={this.onPressSaveButton}
            title="Save"
          />
        </View>
      </View>
    )
  }
}

const createNewFlashcard = gql`
  mutation submitNewFlashcard(
    $name: String!
    $meaning: String!
    $example: String!
    $mastered: Boolean!
  ) {
    createFlashcard(
      name: $name
      meaning: $meaning
      example: $example
      mastered: $mastered
      userId: "cje8649pvb3u201775435vabn"
    ) {
      id
      name
      meaning
      example
      mastered
    }
  }
`

export default graphql<
  {
    id: string
    name: string
    meaning: string
    example: string
    mastered: boolean
  },
  IProps
>(createNewFlashcard)(AddFlashCardScreen)
