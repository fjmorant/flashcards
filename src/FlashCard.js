/* @flow */

import {observable, action} from 'mobx'
import shortid from 'shortid'

export default class FlashCard {
    id = shortid.generate()

    @observable name = ''
    @observable meaning = ''
    @observable example = ''
    @observable mastered = false

    name: string
    meaning: string
    example: string
    mastered: boolean

    @action.bound
    changeName(name: string) {
        this.name = name
    }

    @action.bound
    changeMeaning(meaning: string) {
        this.meaning = meaning
    }

    @action.bound
    changeExample(example: string) {
        this.example = example
    }

    @action.bound
    toggleMaster() {
        this.mastered = !this.mastered
    }
}