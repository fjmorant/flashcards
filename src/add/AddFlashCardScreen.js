import React, {Component, PropTypes} from 'react'
import {
  Button as ReactNativeButton,
  StyleSheet,
  View,
} from 'react-native'
import Input from '../common/Input'
import InputArea from '../common/InputArea'
import Button from '../common/Button'
import {
  changeFlashCardMastered,
  changeFlashCardId,
  changeFlashCardName,
  changeFlashCardMeaning,
  changeFlashCardExample,
  saveFlashCard,
  onSearchEntryTriggered,
  onSelectMeaningOption,
  onSetFlashCardAsMastered,
  onSetFlashCardAsUnmastered,
} from './eventHandlers'
import {clearFlashCard} from './actionCreators'
import {connect} from 'react-redux'
import ModalPicker from '../common/ModalPicker'
import immutable from 'immutable'
import request from 'superagent'

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 8,
  },
  input: {
    margin: 4,
    flex: 1,
    fontSize: 19,
  },
});

class AddFlashCardScreen extends Component {

  static navigationOptions = {
    title: 'Add Flash Cards',
    header: {
      right: null,
    }
  }

  constructor(props) {
    super()

    this.toggleMasterFlashcard = this.toggleMasterFlashcard.bind(this)
    this.changeName = this.changeName.bind(this)
    this.changeMeaning = this.changeMeaning.bind(this)
    this.changeExample = this.changeExample.bind(this)
    this.onSearchEntryTriggered = this.onSearchEntryTriggered.bind(this)
    this.onSelectMeaningOption = this.onSelectMeaningOption.bind(this)
    this.clearPickerOptions = this.clearPickerOptions.bind(this)

    const {state:{params}} = props.navigation

    let flashcard = null

    if (params && params.id) {
      flashcard = props.flashcards.find((flashcard) => flashcard.get('id') === params.id)
    } else {
      flashcard = immutable.fromJS({
        name: '',
        meaning: '',
        example: '',
        mastered: false,
      })
    }

    this.state = {
      flashcard,
      modalVisible: false,
      pickerOptions: immutable.List(),
    }
  }

  shouldEnableSaveButton() {
    return this.state.flashcard.get('name') && this.state.flashcard.get('meaning') && this.state.flashcard.get('example')
  }

  toggleMasterFlashcard() {
    const {state:{params}} = this.props.navigation

    this.props.changeFlashCardMastered(!this.props.mastered)
    this.props.saveFlashCard(params.id)
    this.props.navigation.goBack()
  }

  changeName(name) {
    this.setState({
      flashcard: this.state.flashcard.set('name', name)
    })
  }

  changeMeaning(meaning) {
    this.setState({
      flashcard: this.state.flashcard.set('meaning', meaning)
    })
  }

  changeExample(example) {
    this.setState({
      flashcard: this.state.flashcard.set('example', example)
    })
  }

  onSearchEntryTriggered() {
    if (this.state.flashcard.get('name')) {
      const name = this.state.flashcard.get('name').toLowerCase()

      request
      .get(`https://wordsapiv1.p.mashape.com/words/${name}`)
      .set('X-Mashape-Key', 'VQa8Eayg2Kmsh0nklI2QUhfltYkvp19WrI8jsnxrvGVatb8Lf8')
      .set('Accept', 'application/json')
      .end((error, response) => {
        if (error) {
          Alert.alert(error.message)
          return
        }

        if (response.body.results.length === 1) {
          const entryDictionary = response.body.results[0]
          this.changeMeaning(entryDictionary.definition)
          if (entryDictionary.examples && entryDictionary.examples[0]) {
            this.changeExample(entryDictionary.examples[0])
          }
        } else {
          this.setState({
            pickerOptions: immutable.fromJS(response.body.results),
            modalVisible: true,
          })
        }
      })
    }
  }

  onSelectMeaningOption(entry) {
    const entryDictionary = this.state.pickerOptions.get(entry.key)
    const hasExample = entryDictionary.has('examples') && entryDictionary.getIn(['examples', 0])

    this.setState({
      flashcard: this.state.flashcard.merge({
        meaning: entryDictionary.get('definition'),
        example: hasExample ? entryDictionary.getIn(['examples', 0]) : ''
      }),
      pickerOptions: immutable.List(),
    })
  }

  clearPickerOptions() {
    this.setState({
      pickerOptions: immutable.List(),
      modalVisible: false,
    })
  }

  render() {
    const {state:{params = {}}} = this.props.navigation

    return (
      <View style={styles.container}>
        {
          this.state.pickerOptions.size > 0 ?
              <ModalPicker
                data={this.state.pickerOptions.toJS().map((result, index) => ({
                  key: index,
                  label: result.definition,
                }))}
                initValue="Select part speech"
                modalVisible = {this.state.modalVisible}
                onClose = {this.clearPickerOptions}
                onChange={this.onSelectMeaningOption} /> : null
        }
        <View style={{flexDirection: 'row', alignItems: 'center'}}>
          <Input
              height={35}
              onChangeText={this.changeName}
              placeholder='Type word you want to remember'
              returnKeyType='next'
              style={styles.input}
              value={this.state.flashcard.get('name')}/>
          <ReactNativeButton
            onPress={this.onSearchEntryTriggered}
            title={'Find'}
          />
        </View>
        <InputArea
            onChangeText={this.changeMeaning}
            placeholder='Explanation of the word'
            style={styles.input}
            value={this.state.flashcard.get('meaning')}
        />
        <InputArea
            onChangeText={this.changeExample}
            placeholder='Example of your word'
            style={styles.input}
            value={this.state.flashcard.get('example')}
        />
        <View style={{margin: 5}}>
          <Button
              disabled={!this.shouldEnableSaveButton()}
              height={40}
              onPress={() => {
                this.props.saveFlashCard(this.state.flashcard)
                this.props.navigation.goBack()
              }}
              title='Save'
          />
        </View>
        {
          params.id ?
          <View style={{margin: 5}}>
            <Button
                height={40}
                onPress={this.toggleMasterFlashcard}
                title={this.props.mastered ? 'Mark as Unmastered' : 'Mark as Mastered'}
            />
          </View> : null
        }
      </View>
    );
  }
}

AddFlashCardScreen.propTypes = {
  changeFlashCardExample: PropTypes.func,
  changeFlashCardMeaning: PropTypes.func,
  changeFlashCardName: PropTypes.func,
  example: PropTypes.string,
  meaning: PropTypes.string,
  name: PropTypes.string,
  navigator: PropTypes.object,
  saveFlashCard: PropTypes.func,
}

export default connect(state => ({
    flashcards: state.flashcards.get('flashcards'),
  }),
  (dispatch) => ({
    changeFlashCardMastered: (mastered) => dispatch(changeFlashCardMastered(mastered)),
    changeFlashCardId: (id) => dispatch(changeFlashCardId(id)),
    clearFlashCard: () => dispatch(clearFlashCard()),
    saveFlashCard: (id) => dispatch(saveFlashCard(id)),
  })
)(AddFlashCardScreen)
