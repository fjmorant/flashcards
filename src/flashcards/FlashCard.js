/* @flow */

import {observable, action} from 'mobx'

export default class FlashCard {
    @observable name = ''
    @observable meaning = ''
    @observable example = ''
    @observable mastered = false

    name: string
    meaning: string
    example: string
    mastered: boolean

    @action
    changeName(name) {
        this.name = name
    }

    @action
    changeMeaning(meaning) {
        this.meaning = meaning
    }

    @action
    changeExample(example) {
        this.example = example
    }

    @action
    toggleMaster() {
        this.mastered = !this.mastered
    }
}