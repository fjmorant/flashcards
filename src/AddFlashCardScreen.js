import React, {Component} from 'react'
import {
  Alert,
  Button as ReactNativeButton,
  StyleSheet,
  View,
} from 'react-native'
import Input from './common/Input'
import InputArea from './common/InputArea'
import Button from './common/Button'
import ModalPicker from './common/ModalPicker'
import request from 'superagent'
import {observer} from 'mobx-react/native'
import FlashCard from './FlashCard'
import {graphql} from 'react-apollo'
import gql from 'graphql-tag'

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 8,
  },
  input: {
    margin: 4,
    flex: 1,
    fontSize: 19,
  },
  inputContainer: {flexDirection: 'row', alignItems: 'center'},
  buttonContainer: {margin: 5},
})

@observer(['flashCardList'])
class AddFlashCardScreen extends Component<
  void,
  {
    navigation: any,
    flashCardList: any,
    mastered: boolean,
  },
  void
> {
  static navigationOptions = () => ({
    title: 'Add Flash Cards',
    headerRight: null,
  })

  flashcard = new FlashCard()

  constructor(props) {
    super()

    this.state = {
      modalVisible: false,
      pickerOptions: [],
    }

    if (props.navigation.state.params) {
      this.flashcard = props.flashCardList.getFlashCard(
        props.navigation.state.params.id
      )
    }
  }

  shouldEnableSaveButton() {
    return (
      this.flashcard.name && this.flashcard.meaning && this.flashcard.example
    )
  }

  toggleMasterFlashcard = () => {
    this.flashcard.toggleMaster()
    this.props.flashCardList.edit(this.flashcard)
    this.props.navigation.goBack()
  }

  changeName = (name: string) => {
    this.flashcard.changeName(name)
  }

  changeMeaning = (meaning: string) => {
    this.flashcard.changeMeaning(meaning)
  }

  changeExample = example => {
    this.flashcard.changeExample(example)
  }

  onSearchEntryTriggered = () => {
    if (this.flashcard.name) {
      const name = this.flashcard.name.toLowerCase()

      request
        .get(`https://wordsapiv1.p.mashape.com/words/${name}`)
        .set(
          'X-Mashape-Key',
          'VQa8Eayg2Kmsh0nklI2QUhfltYkvp19WrI8jsnxrvGVatb8Lf8'
        )
        .set('Accept', 'application/json')
        .end((error, response) => {
          if (error) {
            Alert.alert(error.message)
            return
          }

          if (response.body.results.length === 1) {
            const entryDictionary = response.body.results[0]
            this.changeMeaning(entryDictionary.definition)
            if (entryDictionary.examples && entryDictionary.examples[0]) {
              this.changeExample(entryDictionary.examples[0])
            }
          } else {
            this.setState({
              pickerOptions: response.body.results,
              modalVisible: true,
            })
          }
        })
    }
  }

  onSelectMeaningOption = entry => {
    const entryDictionary = this.state.pickerOptions[entry.key]
    const hasExample = entryDictionary.examples && entryDictionary.examples[0]

    this.flashcard.changeMeaning(entryDictionary.definition)
    this.flashcard.changeExample(hasExample ? entryDictionary.examples[0] : '')

    this.setState({
      pickerOptions: [],
    })
  }

  clearPickerOptions = () => {
    this.setState({
      pickerOptions: [],
      modalVisible: false,
    })
  }

  onPressSaveButton = () => {
    this.props
      .mutate({
        variables: {
          name: this.flashcard.name,
          meaning: this.flashcard.meaning,
          example: this.flashcard.example,
          mastered: false,
        },
      })
      .then(({data}) => {
        this.props.navigation.goBack()
      })
      .catch(error => {
        console.log(error)
        alert('Error saving flashcard')
      })
  }

  render() {
    const {state: {params = {}}} = this.props.navigation

    return (
      <View style={styles.container}>
        {this.state.pickerOptions.size > 0 ? (
          <ModalPicker
              data={this.state.pickerOptions.map((result, index) => ({
              key: index,
              label: result.definition,
            }))}
              initValue="Select part speech"
              modalVisible={this.state.modalVisible}
              onChange={this.onSelectMeaningOption}
              onClose={this.clearPickerOptions}
          />
        ) : null}
        <View style={styles.inputContainer}>
          <Input
              height={35}
              onChangeText={this.changeName}
              placeholder="Type word you want to remember"
              returnKeyType="next"
              style={styles.input}
              value={this.flashcard.name}
          />
          <ReactNativeButton
              onPress={this.onSearchEntryTriggered}
              title={'Find'}
          />
        </View>
        <InputArea
            onChangeText={this.changeMeaning}
            placeholder="Explanation of the word"
            style={styles.input}
            value={this.flashcard.meaning}
        />
        <InputArea
            onChangeText={this.changeExample}
            placeholder="Example of your word"
            style={styles.input}
            value={this.flashcard.example}
        />
        <View style={styles.buttonContainer}>
          <Button
              disabled={!this.shouldEnableSaveButton()}
              height={40}
              onPress={this.onPressSaveButton}
              title="Save"
          />
        </View>
        {params.id ? (
          <View style={styles.buttonContainer}>
            <Button
                height={40}
                onPress={this.toggleMasterFlashcard}
                title={
                this.props.mastered ? 'Mark as Unmastered' : 'Mark as Mastered'
              }
            />
          </View>
        ) : null}
      </View>
    )
  }
}

const createNewFlashcard = gql`
  mutation submitNewFlashcard(
    $name: String!
    $meaning: String!
    $example: String!
    $mastered: Boolean!
  ) {
    createFlashcard(
      name: $name
      meaning: $meaning
      example: $example
      mastered: $mastered
      userId: "cje8649pvb3u201775435vabn"
    ) {
      id
    }
  }
`

export default graphql(createNewFlashcard)(AddFlashCardScreen)
