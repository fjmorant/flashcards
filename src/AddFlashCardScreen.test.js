import 'react-native'
import React from 'react'
import AddFlashCardScreen from './AddFlashCardScreen'
import FlashCard from './FlashCard'

import renderer from 'react-test-renderer'

const flashCardList = {
  list: [new FlashCard()],
  getFlashCard: () => new FlashCard(),
}

describe('AddFlashCardScreen', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <AddFlashCardScreen
            flashCardList={flashCardList}
            navigation={{state: {params: {}}}}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
