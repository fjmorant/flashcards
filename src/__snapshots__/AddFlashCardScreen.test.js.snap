// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddFlashCardScreen renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "padding": 8,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Type word you want to remember"
      returnKeyType="next"
      style={
        Array [
          Object {
            "borderColor": "gray",
            "borderWidth": 1,
            "height": 35,
            "padding": 5,
          },
          Object {
            "flex": 1,
            "fontSize": 19,
            "margin": 4,
          },
        ]
      }
      value=""
    />
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={undefined}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Find
        </Text>
      </View>
    </View>
  </View>
  <TextInput
    allowFontScaling={true}
    multiline={true}
    onChangeText={[Function]}
    placeholder="Explanation of the word"
    returnKeyType="next"
    style={
      Array [
        Object {
          "borderColor": "gray",
          "borderWidth": 1,
          "flex": 1,
          "padding": 5,
        },
        Object {
          "flex": 1,
          "fontSize": 19,
          "margin": 4,
        },
      ]
    }
    value=""
  />
  <TextInput
    allowFontScaling={true}
    multiline={true}
    onChangeText={[Function]}
    placeholder="Example of your word"
    returnKeyType="next"
    style={
      Array [
        Object {
          "borderColor": "gray",
          "borderWidth": 1,
          "flex": 1,
          "padding": 5,
        },
        Object {
          "flex": 1,
          "fontSize": 19,
          "margin": 4,
        },
      ]
    }
    value=""
  />
  <View
    style={
      Object {
        "margin": 5,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 5,
              "justifyContent": "center",
            },
            Object {
              "backgroundColor": "rgba(0, 64, 128, 0.5)",
              "height": 40,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "rgb(255, 255, 255)",
              "fontSize": 20,
            }
          }
        >
          Save
        </Text>
      </View>
    </View>
  </View>
</View>
`;
